### ISSUE TITLE
Allow code minification

### ISSUE THREAD

**Issue by Sir-Prise** [buttplugio/buttplug-js#225]
After building my application I get the error "should NOT have additional properties".

I found out that it propably happens due to the minification of the JS code, because at some point it checks if "o" is in a list of DeviceAdded, DeviceList, DeviceRemoved, ...

After that I found out that this is already described in the [Readme](https://github.com/buttplugio/buttplug-js/tree/master/packages/buttplug#buttplug-and-uglifyterser), but obviously it can't be the permanent solution to disable the minifaction of function/classnames.

**Comment by qdot** [comment]
Permanent solution will be the library moving to WASM as a backend, at which point minification won't really need to be a thing. WONTFIX'ing 'cause I'm def not spending time on webpack/minification work before that happens.

**Comment by qdot** [comment]
Reopening this repo and this is definitely gonna be a problem again. Luckily this time I'm not being so weird about trying to build everything out of pure type information, so hopefully I can just add some sort of static value to the message classes and identify by that instead of trying to use class/constructor names.

**Comment by qdot** [comment]
Class/construction names are still hosed here and are breaking in v3. Oops.

**Comment by Sofymu** [comment]
Ran into this as well, had to add the class names to my vite config and switch to terser instead of esbuild as it doesn't support keeping class names non-globally.

```ts
import { defineConfig } from "vite";

export default defineConfig({
  build: {
    minify: "terser",

    terserOptions: {
      keep_classnames: /(Ok|Ping|Error|DeviceList|DeviceAdded|DeviceRemoved|RequestDeviceList|StartScanning|StopScanning|ScanningFinished|RequestServerInfo|ServerInfo|StopDeviceCmd|StopAllDevices|\w+Cmd|SensorReading|RawReading)/
    },
  },
});
```

Honestly, for a temporary solution, giving them a detectable prefix like ButtplugMessageX for example, so we can filter it out with just a single regex entry might just be fine.

**Comment by nickmccurdy** [comment]
Probably worth noting as well that Vite's terser support is no longer the default, requires an extra dependency, and is much slower.

**Comment by qdot** [comment]
Yeah this is Just Bad On A Bunch Of Levels. I'm gonna see about switching to some sort of static naming systems that won't blow up on minification.

**Comment by qdot** [comment]
Fixed in v3.1.0 (releasing to npm shortly). Thanks to @smlt for finally getting this done!
